plasmapy/physics/distribution.py

```diff
@@ -70,25 +77,286 @@ def Maxwellian_1D(v: units.m/units.s,
+    the distribution of particles with speed v in a plasma with temperature T
+    is given by:
 
-    T = k_B*T
+    .. math::
+    f = (\pi * v_Th^2)^{-3/2} \exp(-(\vec{v} - \vec{v_{drift}})^2 / v_Th^2)
+    where v_Th = \sqrt(2 k_B T / m) is the thermal speed
 
-    f = np.sqrt((m_s/(2*pi*T)))*np.exp(-m_s/(2*T)*(v-V_drift)**2)
+    See also
+    --------
+    Maxwellian_1D
+
+    Example
+    -------
+    
+    
+    """
+    # convert temperature to Kelvins
+    T = T.to(units.K, equivalencies=units.temperature_energy())
+    # get thermal velocity and thermal velocity squared
+    vTh = thermal_speed(T, particle=particle, method="most_probable")
+    # accounting for thermal velocity in 3D
+    vTh3D = np.sqrt(3) * vTh
+    vThSq = vTh3D ** 2
+    # Get particle velocity squared
+    vSq = ((vx-vx_drift) ** 2 + (vy-vy_drift) ** 2 + (vz-vz_drift) ** 2)
+    # calculating distribution function
+    try:
+        coeff = (vThSq * np.pi) ** (-3 / 2)
+        expTerm = np.exp(-vSq / vThSq)
+        distFunc = coeff * expTerm
+    except Exception:
+        raise ValueError("Unable to get distribution function.")
+    return distFunc.to((units.s/units.m) ** 3)
+
+@units.quantity_input
+def Maxwellian_speed_1D(v: units.m/units.s,
+                        T, 
+                        particle="e",
+                        v_drift=0*units.m/units.s):
+    r"""Return the probability of finding a particle with speed `v` in m/s
+     in an equilibrium plasma of temperature `T` which follows the 
+     Maxwellian distribution function.
+
+    Parameters
+    ----------
+    v: Quantity
+        The speed in units convertible to m/s
+
+    T: Quantity
+        The temperature, preferably in Kelvin
+
+    particle: string, optional
+        Representation of the particle species(e.g., 'p' for protons, 'D+'
+        for deuterium, or 'He-4 +1' for $He_4^{+1}$ : singly ionized helium-4),
+        which defaults to electrons.
+    
+    v_drift: Quantity
+        The drift speed in units convertible to m/s
+
+    Returns
+    -------
+    f : Quantity
+        probability in speed^-1, normalized so that:
+        $\int_{0}^{\infty} f(v) dv = 1}
+
+    Raises
+    ------
+    TypeError
+        The parameter arguments are not Quantities and
+        cannot be converted into Quantities.
 
-    return f.to(units.s/units.m)
+    UnitConversionError
+        If the parameters is not in appropriate units.
+
+    ValueError
+        If the temperature is negative, or the particle mass or charge state
+        cannot be found.
+
+    Notes
+    -----
+    In one dimension, the Maxwellian speed distribution function describing
+    the distribution of particles with speed v in a plasma with temperature T
+    is given by:
+
+    .. math::
+    f(v) = 4 \pi v^2 (\pi * v_Th^2)^{-3/2} \exp(-(v - v_{drift})^2 / v_Th^2)
+    where v_Th = \sqrt(2 k_B T / m) is the thermal speed
+
+
+    Example
+    -------
+    
```

Comment:
Could definitely use some doctests here :)

Reply:
How do you guys run doctests?

Reply:
That documentation update really can't come soon enough, can it? :D I'll try to automate and simplify the process a bit while I'm at it.

If you clone the repo to a directory called PlasmaPy (so you have a `PlasmaPy/plasmapy` directory where the actual repository is located, you should be able to run `pytest` or more specifically `pytest plasmapy/physics/distribution.py` from the PlasmaPy  directory and that should, due to seeing the `setup.cgg` file, include doctests.